src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,164 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a parser to parse inputs into actions to be taken.
+ */
+public class Parser {
+    private String bye = "bye";
+    private String list = "list";
+    private String mark = "mark";
+    private String unmark = "unmark";
+    private String todo = "todo";
+    private String deadline = "deadline";
+    private String deadlineBy = "/by";
+    private String event = "event";
+    private String eventAt = "/at";
+    private String delete = "delete";
+    private String space = " ";
+    private String find = "find";
+
+    /**
+     * Parses an input to decode it.
+     *
+     * @param d Duke bot to be instructed.
+     * @param input Input to be parsed.
+     * @param isLoading Boolean to indicate if the Duke bot is loading.
+     */
+    public void parse(Duke d, String input, boolean isLoading) {
+        if (input.equals(this.bye)) {
```

Comment:
Since there are multiple if-else statements to check the input against the possible commands, would a switch statement be better here for readability?

Reply:
Switch statements with the current code structure would be too complicated.
Instead, I will shift the bulk of the processes to private methods of the parser class and have the parser call those methods instead. Apparently, the code was not following the single level of abstraction principle which resulted in a very long parse method.